backend/src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -35,20 +34,14 @@
     @Inject
     EntityManager entityManager;
 
+
+
     @Transactional
     public Endpoint createEndpoint(Endpoint endpoint) {
         // Todo: NOTIF-429 backward compatibility change - Remove soon.
         if (endpoint.getType() == EndpointType.CAMEL) {
             CamelProperties properties = endpoint.getProperties(CamelProperties.class);
 
-            if (endpoint.getSubType() == null && properties.getSubType() == null) {
-                throw new BadRequestException("endpoint.subtype must have a value");
-            }
-
-            if (endpoint.getSubType() != null && properties.getSubType() != null && !properties.getSubType().equals(endpoint.getSubType())) {
-                throw new BadRequestException("endpoint.subtype must be equal to endpoint.properties.subtype. Consider removing endpoint.properties.subtype as it is deprecated");
-            }
```

Comment:
Is the removal of these lines a mistake or do you really mean to delete them?

Reply:
Yes, that is a mistake

Reply:
Actually the code works without them and assumes `endpoint.getSubType()` is valid, so I guess there is no real harm in keeping it removed

Reply:
They were added for validation purposes. The code may work without them but we could process an invalid REST request without that validation. @josejulio since you added that, WDYT?

Reply:
they were added in the interim for camel users that were still sending to `properties.subType` instead of `subType` directly.

There are other changes needed.
The whole `// Todo: NOTIF-429 backward compatibility change - Remove soon.` block should go away and there are other blocks with this comment.

Are you aware of any team creating integrations directly via the API ? or is everyone using the UI directly?

Reply:
Currently Eventing is the only team using the API directly (afaik)